[local_dss_bench] Tool to generate automatic graphs for q/s based on various parameters (#1519)
interuss/monitoring: monitoring/dss_bench/README.md

# dss_bench

> Benchmark DSS performance as a function of deployment parameters.

Reviewer: I think this is important as it establishes the scope of the tool and therefore how much we should look to genericize things -- this will only ever use local DSS instances, right?  If so, I think we should also clarify the top module-level name from `dss_bench` to `local_dss_bench`.  If we aspire to have this eventually benchmark deployed DSS instances also, I think we would make some different architectural decisions.

```suggestion
Benchmark performance of a local DSS deployment as a function of deployment parameters.
```

Author: Yes, that defintly something that is and will only be local.

> For every **context** variant x **test**, it: cleans the local ecosystem,

Reviewer: ```suggestion
For every **context** in every sweep for each **test**, it: cleans the local ecosystem,
```

`make start-locally` with the right env, runs the test against all deployed

> DSS at once, and records q/s + latency percentiles. Output is one plot per

Reviewer: ```suggestion
`make start-locally` with the right env, runs the test against all locally-deployed
DSS instances at once, and records q/s + latency percentiles. Output is one plot per
```

test, q/s and latency vs context.

## Layout

- `config.py` - global settings (node count, image, datastore type, duration, processes).
- `environment.py` - wraps `make start-locally` / `down-locally`, resolves DSS targets.
- `auth.py` - mints Dummy OAuth tokens (audience = DSS hostname).
- `driver.py` - runs `processes` processes per DSS for `duration`, times each call.
- `measure.py` - aggregates into total q/s + median/p95 latency.
- `plot.py` - one PNG per test (q/s + latency, one line per comparison arm).
- `arms.py` - optional comparison arms (image vs image, or datastore vs datastore).
- `run.py` - CLI matrix runner.
- `contexts/` - one file per context.
  from 0 to 100 ms).
- `tests/` - one file per test.

## Run

```bash

> make image            # build the DSS/monitoring images once

Reviewer: Does this build the DSS images?
